### PR TITLE
mysql_install.sh

### PR DESCRIPTION
**Pull request by grooverdan**
just a few options to avoid creating databases/users/grants and then deleting them seconds later.

### CODE REVIEW

**Comment by yosifkit** [comment]
Apologies for not reviewing this earlier :bowing_man:

LGTM

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -140,12 +140,10 @@ docker_create_db_directories() {
...
-	installArgs=( --datadir="$DATADIR" --rpm )
-	if { mysql_install_db --help || :; } | grep -q -- '--auth-root-authentication-method'; then
-		# beginning in 10.4.3, install_db uses "socket" which only allows system user root to connect, switch back to "normal" to allow mysql root without a password
-		# see https://github.com/MariaDB/server/commit/b9f3f06857ac6f9105dc65caae19782f09b47fb3
-		# (this flag doesn't exist in 10.0 and below)
-		installArgs+=( --auth-root-authentication-method=normal )
+	installArgs=( --datadir="$DATADIR" --rpm --auth-root-authentication-method=normal )
+	if { mysql_install_db --help || :; } | grep -q -- '--skip-test-db'; then
+                # 10.3+
+		installArgs+=( --skip-test-db )
```
Comment (tianon) [inline docker-entrypoint.sh:146]: Whitespace appears to be mismatched here:

```suggestion
		# 10.3+
		installArgs+=( --skip-test-db )
```

(Tabs vs spaces)

**Comment by grooverdan** [comment]
I'm still working on getting a version of https://github.com/MariaDB/server/pull/1657 upstream to simplify the whole mysql_install_db interface. It will be 10.2, so if you want to hold off on these changes for something a little more invasive (as a simplification too) then by all means you can ignore/close this.

Code freeze is next Tuesday with release sometime afterwards. This will be the last 10.1 release.

**Comment by tianon** [comment]
(Force pushed for rebase and to apply 05be652 across all files :sweat_smile:)

**Pull request merged**